x/internal/outline/outline_packet_proxy.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2023 Jigsaw Operations LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package outline
+
+import (
+	"context"
+	"fmt"
+	"net"
+	"strconv"
+
+	"github.com/Jigsaw-Code/outline-sdk/network"
+	"github.com/Jigsaw-Code/outline-sdk/network/dnstruncate"
+	"github.com/Jigsaw-Code/outline-sdk/transport"
+	"github.com/Jigsaw-Code/outline-sdk/transport/shadowsocks"
+	"github.com/Jigsaw-Code/outline-sdk/x/connectivity"
+)
+
+type OutlinePacketProxy interface {
```

Comment:
Do we need to expose `NewOutlinePacketProxy`, `NewOutlinePacketListener` and `NewOutlineStreamDialer`?

For VPNs, you just need to expose a `NewOutlineIpDevice(accessKey)`. For apps, perhaps a the stream dialer and a packet dialer make sense.

Reply:
I agree , the only user would be connectivity test. But as you mentioned above, it contains special logic of looping through all IP addresses, so after I reverted the `outline-connectivity` there will be no apps using these, and then I can mark all these `StreamDialer` and `PacketProxy` `private`, and only exposes an `OutlineDevice`.